### PR TITLE
Ensure Setting :default values match the Setting :type; fix user-facing-value for calculated Settings. Cleanup & extra tests for email settings

### PR DESCRIPTION
**Pull request by camsaul**
We don't clearly enforce whether Setting defaults are supposed to be unparsed strings or already-parsed values... it seemed like in most places we had been using already-parsed values (which I think seems sane) but other places weren't.

This PR enforces the constraint that a Setting default value has to agree with its `:type` (actually, its `:tag` class, which is usually determined by `:type`) e.g. an `:integer` Setting `:default` value has to be an integer. This constraint is enforced at compile/load time (when the `defsetting` form is evaluated when the namespace gets loaded). 

To make this work I made a small tweak to the way the default Settings getters (`get-string`/`get-boolean`/`get-integer`/etc.) work: they only return the default value if it matches that type. (Previously, `get-string` would call `(str ...)` on the default value to convert it back to a string, and then other functions such as `get-csv` would attempt to parse that value, which didn't always work correctly.) `get-boolean` and other upstream functions correctly return the default value *if* it's a boolean (which we now enforce with the aforementioned constraint.)

I updated the types of a few Settings that clearly should have been typed e.g. LDAP port should be an `:integer` setting, not a `:string` one. I think this stuff was `:string` because we originally only had `:string` Settings and they predated the other types.

This PR also fixes the bug in `user-facing-value` we found yesterday for calculated Settings e.g. `is-hosted?`

TODO:

- [x] I think we need to update a few `defsetting` definitions that had the wrong `:default` type
- [x] We should add tests that `user-facing-value` works correctly for a calculated setting e.g. the `is-hosted?` bug we ran in to in the first place
- [x] Need to add some tests for the type validation stuff (e.g. for different error cases)

### CODE REVIEW

**Comment by camsaul** [comment]
This also caught some weird bugs like the default value for `email-smtp-security` being `(deferred-tru "none")` even tho `"none"` in this case was clearly meant as an enum value

File: src/metabase/models/humanization.clj [hunk 1]
```diff
@@ -47,7 +48,7 @@
 ;; simple replaces hyphens and underscores with spaces and capitalizes
 (defmethod name->human-readable-name :simple
   ([s] (name->human-readable-name :simple s))
-  ([_, ^String s]
+  ([_ ^String s]
```
Comment (noahmoss) [inline src/metabase/models/humanization.clj:51]: Maybe change these `s` variable names to something else now that schema is being aliased as `s`?

**Review by noahmoss — APPROVED** [review]
LGTM; seems like the remaining Cypress failure is caused by the email port setting now being an int?

File: src/metabase/models/humanization.clj (hunk 1 above)
Comment (camsaul) [reply src/metabase/models/humanization.clj:51]: If schema is aliased as `s` it's fine, the alias only applies to referring to things from that namespace (e.g. `s/Keyword` or `s/Int`. `s` by itself wouldn't refer to `schema.core`, so it's not overshadowing anything

**Comment by camsaul** [comment]
looks like a false positive in codecov for the FE since I only changed a cypress test.

**Pull request merged**